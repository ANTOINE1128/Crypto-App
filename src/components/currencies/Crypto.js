import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import CryptoCurrencyList from './CryptoCurrencyList';

const Crypto = () => {
  const [searchQuery, setSearchQuery] = useState('');

  const { currencies, isLoading } = useSelector((store) => store.currencies);

  const lower = searchQuery.toLowerCase();

  const filtered = currencies.filter((currency) => currency.symbol.toLowerCase().includes(lower));

  if (isLoading) {
    return <div>Loading current data...</div>;
  }

  return (
    <div>
      <h1 className="title">CryptoCurrency Tracker</h1>
      <div className="search">
        <input
          type="text"
          placeholder="Search coin..."
          className="search-input"
          value={searchQuery}
          onChange={(e) => setSearchQuery(e.target.value)}
        />
      </div>
      <CryptoCurrencyList currencies={filtered} />
    </div>
  );
};

export default Crypto;
