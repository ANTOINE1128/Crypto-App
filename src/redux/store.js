import { configureStore } from '@reduxjs/toolkit';
import currencyReducer from './crypto/currencySlice';

const store = configureStore({
  reducer: {
    currencies: currencyReducer,
  },
});

export default store;
