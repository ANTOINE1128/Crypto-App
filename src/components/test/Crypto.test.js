import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/store'; 
import Crypto from '../currencies/Crypto';

it('test navbar render', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Provider store={store}>
          <Crypto />
        </Provider>
      </MemoryRouter>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
