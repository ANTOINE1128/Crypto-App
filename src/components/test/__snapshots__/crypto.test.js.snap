// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test navbar render 1`] = `
<div>
  <h1
    className="title"
  >
    CryptoCurrency Tracker
  </h1>
  <div
    className="search"
  >
    <input
      className="search-input"
      onChange={[Function]}
      placeholder="Search coin..."
      type="text"
      value=""
    />
  </div>
  <ul
    className="currency-container"
  />
</div>
`;
